src/common/store/synchronization.slice.ts

```diff
@@ -15,17 +15,24 @@ import {
 import type { HandshakeKeyPair, ReduxIPC } from '@types';
 import { safeJSONParse } from '@utils';
 
+export enum SynchronizationTarget {
+  RENDERER = 'RENDERER',
+  MAIN = 'MAIN',
+  SIGNING = 'SIGNING'
+}
+
 interface SynchronizationState {
   publicKey?: string;
   privateKey?: string;
 
-  isHandshaken: boolean;
+  isHandshaken: Partial<Record<SynchronizationTarget, boolean>>;
```

Comment:
Why is this `Partial`? Can´t it be an object with the value set to `false` by default?

Reply:
It can, but is there a reason to store your own value that'll always be `false`? Since you can't handshake with yourself